script.js

```diff
@@ -1,4 +1,79 @@
+//include current gamemode
+var currentGameMode = 'insert username to begin';
+//include option for username
+var userName = '';
+// track number of games user wins
+var numUserWin = 0;
+// track number of games user lost
+var numUserLost = 0;
+//track number of games user draws
+var numUserDraw = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  // first game mode is when username is keyed in
+  if (currentGameMode == "insert username to begin") {
+    userName = input;
+    currentGameMode = "commence scissors paper stone game";
+    myOutputValue = "hello " + input + " you can now begin the game.";
+// second game mode is when username is not keyed in 
+  } else if (!input) {
+    currentGameMode = "insert username to begin";
+    userName = !input;
+    myOutputValue = "error please insert username to begin";
+  }
   return myOutputValue;
 };
+
+// assign variables to scissors, paper, stone
+var SCISSORS = 'scissors';
+var PAPER = 'paper';
+var STONE = 'stone';
+
+// get a random whole number between 0 to 2
+var getRandomInteger = function () {
+  var randomDecimal = Math.random() * 3;
+  var randomNum = Math.floor(randomDecimal);
+  return randomNum;
+}
+
+// assign random number to scissors, paper, or stone 
+var genComputerChoice = function (randomNum) {
+  if (randomNum == 0) {
+    return SCISSORS;
+  }
+  if (randomNum == 1) {
+    return PAPER;
+  }
+  if (randomNum == 2) {
+    return STONE;
+  }
+
+  // invoke functions
+var computerChoice = genComputerChoice(getRandomInteger());
```

Comment:
var computerChoice needs to be inside main() so it is called ea time main() is called, so a diff computerChoice can be created ea time. Otherwise, computerChoice is only created once, and computer will always play the same symbol...